scripts/src/public/services/orders.ts

```diff
@@ -520,3 +520,40 @@ function checkIfTimedOut(order: db.Order): Promise<void> {
 function getLockResource(type: "create" | "get", ...ids: string[]): string {
 	return `locks:orders:${ type }:${ ids.join(":") }`;
 }
+
+export async function createCrossAppOrder(recipientWalletAddress: string, appId: string, title: string, description: string, amount: number, sender: User, senderDeviceId: string): Promise<OpenOrder> {
+
+	logger().info("creating a cross app order")
+
+	const senderWallet = (await sender.getWallets(senderDeviceId)).lastUsed();
+	if (!senderWallet) {
+		throw UserHasNoWallet(sender.id, senderDeviceId);
+	}
+
+	if (await WalletApplication.getBlockchainVersion(senderWallet.address) !==
+		await WalletApplication.getBlockchainVersion(recipientWalletAddress)) {
+		logger().warn("failed cross app order creation due to blockchain version mismatch");
+		throw UserHasNoWallet(sender.id);
+	}
+
+	const order = db.CrossAppOrder.new({
+		offerId: `cross-app_${sender.appId}_to_${appId}`,
```

Comment:
Can this be generated in the `new()` function internally?

Reply:
I'll need to create a special type to contain both app ids and then override the default behavior, is it worth it? what more, not sure the model needs to be aware to app ids etc, otherwise we wouldnt have create a "composite" id rather save the data items separately. what do you think?